packages/utils/src/ast-utils/helpers.ts

```diff
@@ -26,9 +26,7 @@ export const isNodeOfTypeWithConditions = <
 ): ((
   node: TSESTree.Node | null | undefined,
 ) => node is TSESTree.Node & { type: NodeType } & Conditions) => {
-  const entries = Object.entries(conditions) as ObjectEntries<
-    TSESTree.Node & { type: NodeType }
-  >;
+  const entries = Object.entries(conditions) as ObjectEntries<TSESTree.Node>;
```

Comment:
I have no idea to fix type errors without removing `& { type: NodeType }`...

Reply:
This looks fine to me!